src/TestSuite/Fixture/SchemaStrategyInterface.php

```diff
@@ -0,0 +1,30 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\TestSuite\Fixture;
+
+/**
+ * SchemaStrategyInterface
+ */
+interface SchemaStrategyInterface
+{
+    /**
+     * Loads schema required by fixtures.
+     *
+     * @return void
+     */
+    public function load(): void;
```

Comment:
For migrations would all the configuration options go into the constructor? Currently Migrations is using a `migrate()` function to do the bulk of its work, but we could change that to fit this interface instead.

I also wonder if we *need* to formalize this. My thinking with the schema loaders in 4.next was that at the end of `tests/bootstrap.php` your application's schema should be loaded into the test databases. How that happens is up to you. We provide a path for loading SQL dumps, running migrations, or creating indexes in elasticsearch but there is no formal interface.

A more formal interface could make using a variety of datasources/implementations easier to swap out but I'm not convinced that scenario would come up in practice often enough that we should be designing around it.